services/exporters/exporter.go

```diff
@@ -0,0 +1,37 @@
+package exporters
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+
+	"github.com/mailbadger/app/entities"
+	"github.com/mailbadger/app/logger"
+)
+
+// Exporter represents type for creating exporters for different resource
+type Exporter interface {
+	Export(context.Context, *entities.Report) error
+}
+
+func NewExporter(resource string, s3 s3iface.S3API) Exporter {
```

Comment:
NewExporter can return an error, so instead of creating a default exporter that does nothing, we can return an error saying that the `resource` is unknown.